spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/data/neo4j/ReactiveDataNeo4jTest.java

```diff
@@ -0,0 +1,111 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.test.autoconfigure.data.neo4j;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import org.springframework.boot.autoconfigure.ImportAutoConfiguration;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.core.AutoConfigureCache;
+import org.springframework.boot.test.autoconfigure.filter.TypeExcludeFilters;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.annotation.AliasFor;
+import org.springframework.core.env.Environment;
+import org.springframework.test.context.BootstrapWith;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+/**
+ * Annotation that can be used for a Neo4j test that focuses <strong>only</strong> on
+ * reactive Neo4j components. While {@link DataNeo4jTest @DataNeo4jTest} can be used with
+ * both imperative and reactive repositories, {@code @ReactiveDataNeo4jTest} enables
+ * <strong>only</strong> reactive infrastructure by default!
+ * <p>
+ * Using this annotation will disable full auto-configuration and instead apply only
+ * configuration relevant to Neo4j tests.
+ * <p>
+ * By default, tests annotated with {@code @ReactiveDataNeo4jTest} will use an embedded
+ * in-memory Neo4j process (if available). You must use Neo4j test harness 4.0+ with this
+ * annotation here. That requires a manual upgrade of the dependency.
+ * <p>
+ * When using JUnit 4, this annotation should be used in combination with
+ * {@code @RunWith(SpringRunner.class)}.
+ *
+ * @author Michael J. Simons
+ * @since 2.4.0
+ */
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@Inherited
+@BootstrapWith(ReactiveDataNeo4jTestContextBootstrapper.class)
+@ExtendWith(SpringExtension.class)
+@OverrideAutoConfiguration(enabled = false)
+@TypeExcludeFilters(ReactiveDataNeo4jTypeExcludeFilter.class)
+@AutoConfigureCache
+@AutoConfigureDataNeo4j
+@ImportAutoConfiguration
+@Import(ReactiveCapableNeo4jVersionAssertion.class)
+public @interface ReactiveDataNeo4jTest {
```

Comment:
None of the other store has a reactive specific variant so I am not keen to introduce that. Perhaps the need of a specific annotation surfaces another problem?

Reply:
I will target this together with the reactive `AutoConfiguration` suggestion in the next iteration.

Reply:
Just checked this one and the reasons why we have chosen to introduce a parallel annotation for reactive is some convenience features:
Fail on database < 4.0 (https://github.com/spring-projects/spring-boot/pull/22299/files/b7fa3d1faca912a7c384ec3c20ca93a8fb3b21a8#diff-31539ed0ee3ec3e7c3a490aaa41e5db5R50)
and setting the `spring.data.neo4j.repositories.type` to reactive (https://github.com/spring-projects/spring-boot/pull/22299/files/b7fa3d1faca912a7c384ec3c20ca93a8fb3b21a8#diff-31539ed0ee3ec3e7c3a490aaa41e5db5R50)
Arguably this might be too less features for introducing this parallel infrastructure.

Reply:
None of the other stores have such a variant and I am not keen to introduce it as part of the upgrade. We can reconsider once the upgrade is done of course.